motors/tacho_motor_class.c

```diff
@@ -464,12 +467,7 @@ static ssize_t state_show(struct device *dev, struct device_attribute *attr,
 	int ret, i;
 	size_t size = 0;
 
-	ret = tm->ops->get_state(tm->context);
-	if (ret < 0)
-		return ret;
-
-	if (tm->ramping)
-		ret |= BIT(TM_STATE_RAMPING);
+	ret = tacho_motor_get_state(tm);
```

Comment:
Still need to check for `ret < 0` here and return, otherwise we will get garbage out.